### PR TITLE
doc: changelogg update for mcuboot upmerges.

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added list of relevant changes introduced by MCUBoot synchronization
with 6ec2ec3 and then 5a6e181.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -203,6 +203,23 @@ The following list summarizes the most important changes inherited from upstream
   * Removed all nRF-specific aliases to particular hardware peripherals, because they are no longer needed now that nodes can be addressed by node labels.
     For example, you should now use ``DT_NODELABEL(i2c0)`` instead of ``DT_ALIAS(i2c_0)``.
 
+* Mcuboot:
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:206]:
```suggestion
* MCUboot:
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -203,6 +203,23 @@ The following list summarizes the most important changes inherited from upstream
...
+
+  * Fix erase of possible trailer leftover after reset
+  * Added option to relocate interrupts to application
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:209]: Can you give this option's name, to make it easier for users to find?

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -203,6 +203,23 @@ The following list summarizes the most important changes inherited from upstream
...
+  * Fix single image compilation with serial recovery
+  * Allow single image applications
+  * Allow disable cryptographic check of the image
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:212]: Any references to documentation or Kconfig options?

File: doc/nrf/releases/release-notes-latest.rst [hunk 4]
```diff
@@ -203,6 +203,23 @@ The following list summarizes the most important changes inherited from upstream
...
+  * serial_recovery: Reduce minimum number of members in packet
+  * Introduced direct execute-in-place (XIP) mode
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:214]: Same here, reference?

File: doc/nrf/releases/release-notes-latest.rst [hunk 5]
```diff
@@ -203,6 +203,23 @@ The following list summarizes the most important changes inherited from upstream
...
+  * Fix kramdown CVE-2020-14001
+  * Allow to point zephyr project related key via Kconfig (Use private key)
+  * boot: Support for nrf52840 with ecc keys and cryptocell
+  * Allow to set VTOR to relay vector before chain-load the application
+  * Allow larger primary slot in swap-move
```
Comment (mbolivar-nordic) [inline doc/nrf/releases/release-notes-latest.rst:219]: Any references?

Comment (nvlsianpu) [reply doc/nrf/releases/release-notes-latest.rst:219]: This was not patch to the move algo, but to before check. Added more description on that.
https://github.com/JuulLabs-OSS/mcuboot/pull/760

**Review by mbolivar-nordic — COMMENTED** [review]
Nice, thanks for the references!

I think a doc maintainer should confirm, but just FYI, I think the usual style is to write:

```
:option:`CONFIG_FOO`
        ^ no space here
```

Instead of:

```
:option: `CONFIG_FOO`
        ^ space here
```

**Comment by nvlsianpu** [comment]
^^ Right, "no space" is rather used elsewhere.

**Review by ru-fu — CHANGES_REQUESTED** [review]
I'll add a commit with language updates.

**Comment by ru-fu** [comment]
Is 1.6.0-rc2 still the correct version for MCUboot?

File: doc/nrf/releases/release-notes-latest.rst [hunk 6]
```diff
@@ -12,6 +12,29 @@ Changelog
 
 See the `master branch section in the changelog`_ for a list of the most important changes.
 
+
+sdk-mcuboot
+===========
+
+The MCUboot fork in |NCS| contains all commits from the upstream MCUboot repository up to and including ``???``, plus some |NCS| specific additions.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:19]: What commit?

File: doc/nrf/releases/release-notes-latest.rst [hunk 7]
```diff
@@ -12,6 +12,29 @@ Changelog
...
+
+The following list summarizes the most important changes inherited from upstream MCUboot:
+
+  * Fixed an issue where a left-over image trailer was sometimes not erased after reset.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:23]: Did I interpret the input correctly?

File: doc/nrf/releases/release-notes-latest.rst [hunk 8]
```diff
@@ -12,6 +12,29 @@ Changelog
...
+  * Added a ``CONFIG_BOOT_INTR_VEC_RELOC`` option to relocate interrupts to application.
+  * Fixed single image compilation with serial recovery.
+  * Added support for single-image applications (see :option:`CONFIG_SINGLE_IMAGE_DFU`).
+  * Added a ``CONFIG_BOOT_SIGNATURE_TYPE_NONE`` option to disable the cryptographic check of the image.
+  * Reduced the minimum number of members in packet for serial recovery.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:28]: What packet are we talking about? Is that clear enough?

File: doc/nrf/releases/release-notes-latest.rst [hunk 9]
```diff
@@ -12,6 +12,29 @@ Changelog
...
+  * Introduced direct execute-in-place (XIP) mode (see ``CONFIG_BOOT_DIRECT_XIP``).
+  * Fixed kramdown CVE-2020-14001.
+  * Added support for pointing to a Zephyr project related key via Kconfig (use private key).
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:31]: What does "use private key" mean here? Is that the Kconfig option that must be selected?

**Review by ru-fu — COMMENTED** [review]
I added a commit with updates - please check and squash, if okay. There are also some open questions.

It seems there is an issue with some of the Kconfig options not being included in the doc output, so we cannot link to them until this is fixed.

**Comment by nvlsianpu** [comment]
> Is 1.6.0-rc2 still the correct version for MCUboot?
@ru-fu No, MCUBoot is at post 1.6.0 release version.

File: doc/nrf/releases/release-notes-latest.rst (hunk 9 above)
Comment (nvlsianpu) [reply doc/nrf/releases/release-notes-latest.rst:31]: I mean key which is somewhere in a user project directory. @MarekPieta how to describe this better?

File: doc/nrf/releases/release-notes-latest.rst (hunk 8 above)
Comment (nvlsianpu) [reply doc/nrf/releases/release-notes-latest.rst]: smp packet

File: doc/nrf/releases/release-notes-latest.rst (hunk 7 above)
Comment (nvlsianpu) [reply doc/nrf/releases/release-notes-latest.rst:23]: @ru-fu This fixes an issue where an image might be erased, but a trailer left behind.

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:23]:
```suggestion
  * Fixed an issue where after erasing an image, an image trailer might be left behind.
```

**Comment by ru-fu** [comment]
> 
> 
> > Is 1.6.0-rc2 still the correct version for MCUboot?
> > @ru-fu No, MCUBoot is at post 1.6.0 release version.

Then this must be updated in the nrf/doc/mcuboot/conf.py file (line 104 & 106).

**Comment by nvlsianpu** [comment]
^^ will update

File: doc/nrf/releases/release-notes-latest.rst (hunk 9 above)
Comment (MarekPieta) [reply doc/nrf/releases/release-notes-latest.rst:31]: Maybe:
`Modified build system to let application use private key path that begins in user project configuration directory`

Comment (nvlsianpu) [reply doc/nrf/releases/release-notes-latest.rst:31]: @ru-fu ^^ ?

**Comment by utzig** [comment]
> ^^ will update

You can also update the version here: https://github.com/nrfconnect/sdk-nrf/blob/master/doc/versions.json#L17

File: doc/mcuboot/conf.py [hunk 10]
```diff
@@ -101,9 +101,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '1.6.0-rc2'
+version = '1.6.0.99'
```
Comment (nvlsianpu) [inline doc/mcuboot/conf.py:104]: @ru-fu Version 1.6.0 > x< {1.6.1, 1.7.0}   1.6.0.99 or 1.6.99?

Comment (ru-fu) [reply doc/mcuboot/conf.py:104]: No idea. @carlescufi ?

File: doc/nrf/releases/release-notes-latest.rst (hunk 9 above)
Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:31]:
```suggestion
  * Modified the build system to let the application use a private key that is located in the user project configuration directory.
```

File: doc/mcuboot/conf.py (hunk 10 above)
Comment (carlescufi) [reply doc/mcuboot/conf.py:104]: This should be `1.6.99`.

**Comment by ru-fu** [comment]
> 
> 
> > ^^ will update
> 
> You can also update the version here: https://github.com/nrfconnect/sdk-nrf/blob/master/doc/versions.json#L17

Can you also update that one (actually line 18 now)? We need both.

**Comment by nvlsianpu** [comment]
^^ @ru-fu  done, zephyr version updated in zephyr changelogg PR.

**Review by ru-fu — APPROVED** [review]
Looks good to me! :)

**Comment by mbolivar-nordic** [comment]
Overriding unrelated CI errors.

**Pull request merged**